iceoryx_hoofs/test/moduletests/test_shared_memory.cpp

```diff
@@ -100,11 +104,105 @@ TEST_F(SharedMemory_Test, MoveCTorWithValidValues)
 
 TEST_F(SharedMemory_Test, getHandleOfValidObject)
 {
-    auto sut = iox::posix::SharedMemory::create("/ignatz..",
-                                                iox::posix::AccessMode::READ_WRITE,
-                                                iox::posix::OwnerShip::MINE,
-                                                S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH,
-                                                128);
-    EXPECT_THAT(sut->getHandle(), Ne(-1));
+    auto sut = createSharedMemory(SUT_SHM_NAME, iox::posix::OpenMode::PURGE_AND_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, UnlinkNonExistingShmFails)
+{
+    auto result = iox::posix::SharedMemory::unlinkIfExist("/look_there's_a_dead_seagull_flying_its_name_is_dietlbart");
+    ASSERT_FALSE(result.has_error());
+    EXPECT_FALSE(*result);
+}
+
+TEST_F(SharedMemory_Test, UnlinkExistingShmWorks)
+{
+    constexpr const char SHM_NAME[] = "/its_a_mee_monukulius";
+    ASSERT_FALSE(iox::posix::posixCall(iox_shm_open)(SHM_NAME, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP)
+                     .failureReturnValue(SharedMemory::INVALID_HANDLE)
+                     .evaluate()
+                     .has_error());
+    auto result = iox::posix::SharedMemory::unlinkIfExist(SHM_NAME);
+    ASSERT_FALSE(result.has_error());
+    EXPECT_TRUE(*result);
 }
+
+TEST_F(SharedMemory_Test, ExclusiveCreateWorksWhenShmDoesNotExist)
+{
+    IOX_DISCARD_RESULT(iox::posix::SharedMemory::unlinkIfExist(SUT_SHM_NAME));
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::EXCLUSIVE_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, ExclusiveCreateFailsWhenShmExists)
+{
+    ASSERT_TRUE(createRawSharedMemory(SUT_SHM_NAME));
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::EXCLUSIVE_CREATE);
+    ASSERT_TRUE(sut.has_error());
+    IOX_DISCARD_RESULT(iox::posix::SharedMemory::unlinkIfExist(SUT_SHM_NAME));
+}
+
+TEST_F(SharedMemory_Test, PurgeAndCreateWorksWhenShmDoesNotExist)
+{
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::PURGE_AND_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, PurgeAndCreateWorksWhenShmExists)
+{
+    ASSERT_TRUE(createRawSharedMemory(SUT_SHM_NAME));
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::PURGE_AND_CREATE);
+    ASSERT_FALSE(sut.has_error());
+    EXPECT_TRUE(sut->hasOwnership());
+    EXPECT_THAT(sut->getHandle(), Ne(SharedMemory::INVALID_HANDLE));
+}
+
+TEST_F(SharedMemory_Test, CreateOrOpenCreatesShmWhenShmDoesNotExist)
+{
+    auto sut = createSharedMemory(SUT_SHM_NAME, OpenMode::OPEN_OR_CREATE);
```

Comment:
I guess here should also `unlinkIfExists` be called beforehand ... or maybe in `Setup` for `SUT_SHM_NAME`